oximeter/db/src/oxql/query/mod.rs

```diff
@@ -362,6 +366,32 @@ impl Query {
     pub(crate) fn parsed_query(&self) -> &QueryNode {
         &self.parsed
     }
+
+    pub(crate) fn add_filters(&self, filters: Vec<(String, Uuid)>) -> Self {
+        let mut new_ops = self.parsed.table_ops().cloned().collect::<Vec<_>>();
+        let filters = filters
+            .iter()
+            .map(|(key, value)| {
+                let simple_filter = SimpleFilter {
+                    ident: Ident(key.to_string()),
+                    cmp: Comparison::Eq,
+                    value: Literal::Uuid(value.clone()),
+                };
+                let filter_expr = FilterExpr::Simple(simple_filter);
+                let filter_op = Filter { negated: false, expr: filter_expr };
+                TableOp::Basic(BasicTableOp::Filter(filter_op))
```

Comment:
There are some private helpers that would make this shorter. Might want to make them `pub(crate)`.